web/components/contract/contract-report-resolution.tsx

```diff
@@ -0,0 +1,57 @@
+import { Contract } from 'common/contract'
+import { useUser } from 'web/hooks/use-user'
+import clsx from 'clsx'
+import { updateContract } from 'web/lib/firebase/contracts'
+import { Tooltip } from '../tooltip'
+import { ConfirmationButton } from '../confirmation-button'
+import { Row } from '../layout/row'
+import { FlagIcon } from '@heroicons/react/solid'
+
+export function ContractReportResolution(props: { contract: Contract }) {
+  const { contract } = props
+  const user = useUser()
+  if (!user) {
+    return <></>
+  }
+  const userReported = contract.resolutionReports?.includes(user.id)
+
+  const onSubmit = async () => {
+    if (!user) {
+      return true
+    }
+    await updateContract(contract.id, {
+      resolutionReports: [...(contract.resolutionReports || []), user.id],
```

Comment:
you can use my convenience function: bulidArray(contract.resolutionReports, user.id)